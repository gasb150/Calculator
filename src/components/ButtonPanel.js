import React, { Fragment } from 'react';
import Button from './Button';
import '../style/ButtonPanel.css';

const ButtonPanel = () => (
  <>
    <div className="d-grid">
      <Button buttonName="AC" type="sp" />
      <Button buttonName="+/-" type="sp" />
      <Button buttonName="%" type="sp" />
      <Button buttonName="/" type="op" />
    </div>
    <div className="d-grid">
      <Button buttonName="7" type="num" />
      <Button buttonName="8" type="num" />
      <Button buttonName="9" type="num" />
      <Button buttonName="X" type="op" />
    </div>
    <div className="d-grid">
      <Button buttonName="4" type="num" />
      <Button buttonName="5" type="num" />
      <Button buttonName="6" type="num" />
      <Button buttonName="-" type="op" />
    </div>
    <div className="d-grid">
      <Button buttonName="1" type="num" />
      <Button buttonName="2" type="num" />
      <Button buttonName="3" type="num" />
      <Button buttonName="+" type="op" />
    </div>
    <div className="d-grid-last">
      <Button buttonName="0" type="num" />
      <Button buttonName="." type="num" />
      <Button buttonName="=" type="op" />

    </div>
  </>
);

export default ButtonPanel;
