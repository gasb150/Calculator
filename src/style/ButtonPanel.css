.display-value{
  border: red 5px solid;
  text-align: end;
}
.d-grid{
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.d-grid-last{
  display: grid;
  grid-template-columns: 1fr 1fr 2fr;
}
.num{
  background-color: orange;
}
.op{
  background-color: greenyellow;
}
.sp{
  background-color: rgb(226, 208, 45);
}