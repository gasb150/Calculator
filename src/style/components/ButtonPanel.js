import React, {Fragment} from 'react'


const ButtonPanel = () =>{ 
  return(
    <Fragment>
    <div>
      AC, +/-, %, /
    </div>
    <div>
      7, 8, 9, X
    </div>
    <div>
      4, 5, 6, -
    </div>
    <div>
      1, 2, 3, +
    </div>
    <div>
      0, ., =
    </div>
    </Fragment>
  )
}

export default ButtonPanel