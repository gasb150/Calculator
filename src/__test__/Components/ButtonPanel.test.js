import { create } from 'react-test-renderer';

import ButtonPanel from '../../components/ButtonPanel';

it('Should render correctly', () => {
  const elem = create(<ButtonPanel clickHandler={() => 'data'} />);
  expect(elem).toMatchSnapshot();
});
