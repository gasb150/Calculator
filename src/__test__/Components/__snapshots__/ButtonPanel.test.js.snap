// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
Array [
  <div
    className="d-grid"
  >
    <button
      className="ui button purple"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="ui button purple"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="ui button purple"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="ui button blue"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>,
  <div
    className="d-grid"
  >
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="ui button blue"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>,
  <div
    className="d-grid"
  >
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="ui button blue"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="d-grid"
  >
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="ui button blue"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="d-grid-last"
  >
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="ui button teal"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="ui button blue"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>,
]
`;
